React-Dojo/react_full_course_2022/src/App.js

```diff
@@ -0,0 +1,48 @@
+import { useState, useEffect } from "react";
+import SearchIcon from "./search.svg";
+import "./App.css";
+import MovieCard from "./MovieCard";
+
+const API_URL = 'http://www.omdbapi.com/?apikey=9fc736b6'
+
+const App = () => {
+
+  const [movies, setMovies] = useState([])
+  const [searchTerm, setSearchTerm] = useState('')
+
+  const searchMovies = async (title) => {
+
+    const response = await fetch(`${API_URL}&s=${title}`)
+    const data = await response.json()
+
+    setMovies(data.Search);
+  }
+
+  useEffect(() => {
+    searchMovies('Spiderman')
+  }, [])
+
+  return (
+    <div className="app">
+      <h1>Movieland</h1>
+      <div className="search">
+        <input placeholder='Search for movies...' value={searchTerm} onChange={(e) => setSearchTerm(e.target.value)} />
+        <img src={SearchIcon} alt="search" onClick={() => searchMovies(searchTerm)} />
```

Comment:
the onClick event handler could be written in a different and more futureproof way

```
#somewhere in the code before the return
const onImageClickHandler = () => searchMovies(searchTerm);

...
<img ... onClick={onImageClickHandler} />
```